### PR TITLE
Sonarqube issues

### PR DESCRIPTION
**Pull request by jansowa**
SonarQube issues fixes.

I tried to make changes only in safe, non-invasive places.

### CODE REVIEW

File: src/main/java/com/jvm_bloggers/ApplicationProfiles.java [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 public class ApplicationProfiles {
 
+    private ApplicationProfiles() {
+        throw new IllegalStateException("Utility class");
+    }
+
```
Comment (goostleek) [inline src/main/java/com/jvm_bloggers/ApplicationProfiles.java:8]: We have a Lombok dependency in the project already. We can use [`@UtilityClass`](https://projectlombok.org/features/experimental/UtilityClass) annotation instead.
```suggestion
@UtilityClass
public class ApplicationProfiles {
```

File: src/main/java/com/jvm_bloggers/core/newsletter_issues/PublishingConstants.java [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 class PublishingConstants {
 
+    private PublishingConstants() {
+        throw new IllegalStateException("Utility class");
+    }
+
```
Comment (goostleek) [inline src/main/java/com/jvm_bloggers/core/newsletter_issues/PublishingConstants.java:8]: We have a Lombok dependency in the project already. We can use [`@UtilityClass`](https://projectlombok.org/features/experimental/UtilityClass) annotation instead.
```suggestion
@UtilityClass
class PublishingConstants {
```

File: src/main/java/com/jvm_bloggers/utils/HtmlEmptinessChecker.java [hunk 3]
```diff
@@ -4,6 +4,10 @@
 
 public class HtmlEmptinessChecker {
 
+    private HtmlEmptinessChecker() {
+        throw new IllegalStateException("Utility class");
+    }
+
```
Comment (goostleek) [inline src/main/java/com/jvm_bloggers/utils/HtmlEmptinessChecker.java:10]: We have a Lombok dependency in the project already. We can use [`@UtilityClass`](https://projectlombok.org/features/experimental/UtilityClass) annotation instead.
```suggestion
@UtilityClass
public class HtmlEmptinessChecker {
```

File: src/main/java/com/jvm_bloggers/utils/DateTimeUtilities.java [hunk 4]
```diff
@@ -14,6 +14,10 @@
 
 public class DateTimeUtilities {
 
+    private DateTimeUtilities() {
+        throw new IllegalStateException("Utility class");
+    }
+
```
Comment (goostleek) [inline src/main/java/com/jvm_bloggers/utils/DateTimeUtilities.java:20]: We have a Lombok dependency in the project already. We can use [`@UtilityClass`](https://projectlombok.org/features/experimental/UtilityClass) annotation instead.
```suggestion
@UtilityClass
public class DateTimeUtilities {
```

Comment (jansowa) [reply src/main/java/com/jvm_bloggers/utils/DateTimeUtilities.java]: Thank you. Done :)

File: src/main/java/com/jvm_bloggers/utils/HtmlEmptinessChecker.java (hunk 3 above)
Comment (jansowa) [reply src/main/java/com/jvm_bloggers/utils/HtmlEmptinessChecker.java]: Thank you. Done :)

File: src/main/java/com/jvm_bloggers/ApplicationProfiles.java (hunk 1 above)
Comment (jansowa) [reply src/main/java/com/jvm_bloggers/ApplicationProfiles.java]: Thank you. Done :)

**Pull request merged**